### ISSUE TITLE
Support for pattern properties

### ISSUE THREAD

**Issue by epiqu1n** [n3ps/json-schema-to-jsdoc#56]
It would be useful if there was support for pattern properties in a schema. Currently, if a property of type "object" uses the "patternProperties" field to represent an object with dynamic properties, the output type is simply "object". For example, given this part of a schema:
```json
"test": {
  "type": "object",
  "patternProperties": {
    ".*": {
      "type": "number"
    }
  }
},
```
The tool just yields
```js
/**
 * @prop {Object<string, number>} test
 */
```

**Comment by n3ps** [comment]
That's an interesting proposal. I see that [pattern properties](https://json-schema.org/understanding-json-schema/reference/object.html#pattern-properties) take in any regex pattern. Since that's open-ended, we should probably limit to predefined patterns, unless there's a converter that already exists.

**Comment by epiqu1n** [comment]
Makes sense. Considering there's no support for regex in JSDoc, I think it would be best to either only support very generic patterns like `.*` or `.+`, or to support any pattern property and, if there are multiple pattern properties at the same level, merge their contents.  For example in the latter case, this property could yield the following doc:
```json
"test": {
  "type": "object",
  "patternProperties": {
    "\\w+": {
      "type": "object",
      "properties": {
        "wordProp": "string",
        "otherProp": "string"
      },
     "\\d+": {
       "type": "object",
       "properties": {
         "numProp": "number",
         "otherProp": "number"
    }
  }
}
```
```js
/**
 * @prop {Object<string, {wordProp?: string, numProp?: number, otherProp: string | number}>} test
 * or
 * @prop {{[key: string]: {wordProp?: string, numProp?: number, otherProp: string | number}}}
 */
```
Again, messy to look at, but still practical for use with things like IntelliSense.

**Comment by brettz9** [comment]
One might optionally allow preserving the key type too by taking such an approach as follows. Using the prior example:

```js
/**
 * PATTERN: /\w+/
 * @typedef {string} patternPropertiesRegex1
 */

/**
 * PATTERN: /\d+/
 * @typedef {string} patternPropertiesRegex2
 */

/**
 * @prop {
 *   Object<patternPropertiesRegex1, {wordProp?: string, otherProp: string}> | 
 *   Object<patternPropertiesRegex2, {numProp?: number, otherProp: number}>
 * } test
 */
```

I'm not a fan of the merging unless as an option as it can lead to not only less readability, but inaccuracy (e.g., suggesting that an object with `patternPropertiesRegex1` can accept `numProp`).

However, merging could be done here too if there were a need:

```js
/**
 * @prop {Object<patternPropertiesRegex1|patternPropertiesRegex2, {wordProp?: string, numProp?: number, otherProp: string|number}>} test
 */
```

**Comment by epiqu1n** [comment]
Ah you're right that is definitely a better and more practical way to represent the key types, good idea
